libhdf5/hdf5var.c

```diff
@@ -697,25 +702,35 @@ nc_def_var_extra(int ncid, int varid, int *shuffle, int *unused1,
 
         /* Set the deflate settings. */
         var->contiguous = NC_FALSE;
+        var->compact = NC_FALSE;
         var->deflate = *deflate;
         if (*deflate)
             var->deflate_level = *deflate_level;
         LOG((3, "%s: *deflate_level %d", __func__, *deflate_level));
     }
 #endif
 
+    /* Cannot set filters of any sort on scalars */
+    if(var->ndims == 0) {
+	if(shuffle && *shuffle)
+	    return NC_NOERR; /* ignore */
```

Comment:
I don't think this is a great idea. It is already disallowed to set filters on scalars, and the result should not NC_EINVAL, not to ignore it.

Reply:
I would agree, silently ignoring behavior might lead the user to think it had succeeded, leading to confusion down the line.